Update terragrunt to latest release 0.12.5

loupgaroublond opened the pull request:
This updates the terragrunt version, sets its sha256 sum and sets a dependency
on the latest Terraform 0.9.4

This is still a WIP with @shlevy 

###### Motivation for this change

Terraform 0.8 has some broken stuff with respect to modules, to use terragrunt with 0.9.x, you've got to update to the bleeding edge latest release.

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

Commit:
Update terragrunt to latest release 0.12.5

This updates the terragrunt version, sets its sha256 sum and sets a dependency
on the latest Terraform 0.9.4

peterhoeg commented:
We need to carry both terragrunt 0.11 and 0.12 for some time as there is some work associated with upgrading from one to the other. Can you split 0.12 out?

shlevy commented:
@loupgaroublond See e.g. `0.9.8.nix` in that directory for an example of keeping an old version around.

Commit:
Add back the older release of terragrunt under a versioned attribute

Copies back the 0.11.1 nix package of Terragrunt into a versioned file
and modifies the all-packages.nix file to refer to it along with its
Terraform dependencies

loupgaroublond commented:
@peterhoeg I've included the older version of Terragrunt. This is still a WIP as I would like to test this in the sandbox properly and follow the process, but any feedback you have for now, I'll be happy to incorporate.

I likely won't have time to set up the sandbox until this weekend, so please standby on that.

shlevy commented on pkgs/top-level/all-packages.nix:
```diff
@@ -18387,6 +18387,10 @@ with pkgs;
   terraform = terraform_0_9;
 
   terragrunt = callPackage ../applications/networking/cluster/terragrunt {
+    terraform = terraform;
```
You don't need the `terraform = terraform` here, you can just do `callPackage ../applications/networking/cluster/terragrunt {}`

Commit:
Fix redundant parameterization of terraform

peterhoeg commented:
@loupgaroublond 

1) instead of carrying 2 files that are almost identical, you can also use a function but it's probably not worth it in this case, as we will be dropping the old version sometime over the next few months. For inspiration you can see how it's done with [terraform](https://github.com/NixOS/nixpkgs/blob/master/pkgs/applications/networking/cluster/terraform/default.nix) or [vala](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/compilers/vala/default.nix).

2) are we sure that the dependencies haven't changed? 0.11 and 0.12 are both using the same ```deps.nix``` - they probably need separate files. I have a very crude script  [here](https://github.com/peterhoeg/nix-upgrade-scripts/blob/master/bin/nix-upgrade-go) that can help with updating dependencies.

loupgaroublond commented:
I'd argue that #1 only has value if there's a desire to support multiple terraform versions long term and provide a neat orderly library of versions. But that's just my opinion.

Re #2, I haven't had any issues using this in our environment at work. I can work out the differences this weekend when I polish it up for the nix community and see if there's anything to update. Thanks for the pointer.

peterhoeg commented:
I agree on point 1 - probably not worth it in this case.

Please squash the commits and rename the PR to follow the current standard (terragrunt: 0.11.1 -> 0.12.15).

loupgaroublond commented:
Do you mean that I should either force push a new head to this branch or submit a new branch for PR? Just looking to clarify

peterhoeg commented:
Just force push to this branch.

joachifm commented:
I took the liberty of squashing this. Thanks.

Mic92 commented:
Upstream test seems to be not working on x686 anymore: https://hydra.nixos.org/build/52549746
